File: Binary_Search.cpp
 #include<iostream>
 using namespace std;
 int binarySearch(int arr[], int size, int key){
    int start = 0;
    int end = size-1;
    int mid= (start+end)/2;

    while(start <= end){
        if(arr[mid]== key){
            return mid;
        }
        if(key>arr[mid]){
            start=mid+1;
        }
        else{
            end=mid-1;
        }
        mid=(start+end)/2;
    }
    return -1;
 }
 int main(){
    int even[6]={2,4,6,8,10,12};
    int odd[6]={1,3,5,9,11,13};
    int index=binarySearch(even,6,12);
    cout<<"index of 12 is"<<index<<endl;

    return 0;
 }

File: Algorithms/BinarySearch.java
public class BinarySearch {
    public static int binarySearch(int[] arr, int NumberSearch) {
        int low = 0;
        int high = arr.length - 1;

        while (low <= high) {
            int mid = (low + high) / 2;

            
            if (arr[mid] == NumberSearch) {
                return mid; 
            }
            
            if (arr[mid] < NumberSearch) {
                low = mid + 1;
            }
            else {
                high = mid - 1;
            }
        }
        return -1; 
    }

    public static void main(String[] args) {

        int[] arr = {10, 3, 51, 7, 9, 11, 13, 15};

        // Element to search for
        int NumberSearch = 7;

        int result = binarySearch(arr, NumberSearch);


        if (result == -1) {
            System.out.println("Element not present in array");
        } else {
            System.out.println("Element found at index " + result);
        }
    }
}
